### PR TITLE
Adds check that no articulation root API is applied on rigid bodies

### PR DESCRIPTION
**Pull request by lgulich**
# Description

If a USD contains an articulation root but we load it with a `RigidObjectCfg` this would also load the articulation. This is confusing as a user only expects a rigid body.

This PR adds a fix to ignore the articulation if a rigid body is spawned.

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Screenshots

Please attach before and after screenshots of the change if applicable.

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Review by Dhoeller19 — APPROVED** [review]

**Comment by Dhoeller19** [comment]
@lgulich could you please update the changelog, mentioning why this was done

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object.py [hunk 1]
```diff
@@ -288,6 +288,17 @@ def _initialize_impl(self):
                 " Please ensure that there is only one rigid body in the prim path tree."
             )
 
+        articulation_prims = sim_utils.get_all_matching_child_prims(
+            template_prim_path, predicate=lambda prim: prim.HasAPI(UsdPhysics.ArticulationRootAPI)
+        )
+        if len(articulation_prims) != 0:
+            raise RuntimeError(
+                f"Found an articulation root when resolving '{self.cfg.prim_path}'."
+                f" Found '{articulation_prims}' under '{template_prim_path}'."
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object.py]:
```suggestion
                f"Found an articulation root when resolving '{self.cfg.prim_path}' for rigid objects."
                f" These are located at: '{articulation_prims}' under '{template_prim_path}'."
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object.py [hunk 2]
```diff
@@ -288,6 +288,17 @@ def _initialize_impl(self):
...
+                " Please disable the articulation root in the USD or from code using"
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object.py]:
```suggestion
                " Please disable the articulation root in the USD or from code by setting"
```

File: source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object.py [hunk 3]
```diff
@@ -288,6 +288,17 @@ def _initialize_impl(self):
...
+                " the ArticulationRootPropertiesCfg when spawning the rigid object."
```
Comment (Mayankm96) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/assets/rigid_object/rigid_object.py]:
```suggestion
                " the parameter 'ArticulationRootPropertiesCfg.articulation_enabled' to False in the spawn configuration."
```

**Review by Mayankm96 — APPROVED** [review]

File: source/extensions/omni.isaac.lab/docs/CHANGELOG.rst [hunk 4]
```diff
@@ -1,6 +1,14 @@
 Changelog
 ---------
 
+0.27.14 (2024-11-04)
+~~~~~~~~~~~~~~~~~~~~
+
+* If a USD that contains an articulation root is loaded using a
+  :attr:`omni.isaac_lab.assets.RigidBody` we now fail unless the articulation root is explicitly
+  disabled.
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab/docs/CHANGELOG.rst]:
```suggestion
  disabled. Using an articulation root for rigid bodies is not needed and decreases overall performance.
```

**Pull request merged**